### ISSUE TITLE
New German Data Set Thorsten – 2022.10 Available for Re-Training

### ISSUE THREAD

**Issue by eqikkwkp25-cyber** [NVIDIA/NeMo#5400]
Are there any plans to re-train and provide new models based on the new German data set Thorsten 2022.10
which can be found here [https://www.thorsten-voice.de/en/datasets-2/)](https://www.thorsten-voice.de/en/datasets-2/)

Thanks.

**Comment by XuesongYang** [comment]
This new model will apply our latest IPAtokenizer and consider a mixer token representation of graphemes and phonemes. I will link the PR here once it is done.

**Comment by thorstenMueller** [comment]
Nice, thanks for that 😊. If i can support on that, please let me know.

**Comment by XuesongYang** [comment]
@thorstenMueller one question about your new 22.10 neutral dataset. It seems supplementary to 21.02 (equivalent to openslr-95). **Is my understanding correct**? If so, we'd like to pull the two datasets together for training models. Thanks.

**Comment by thorstenMueller** [comment]
@XuesongYang Honestly, i would not combine these two datasets. I recorded 22.10 because i was not satisfied with the 21.02 version. So the new dataset has been recorded with a better recording setup (recording chamber and better microphone). So audio quality is better and the new dataset has a way more natural speechflow and higher character per second speech speed.
But technically you could combine them, but i'm not quite sure what type of result it will be.

**Comment by XuesongYang** [comment]
make senses! Thanks for the info!

**Comment by thorstenMueller** [comment]
Not sure if this topic is still interesting for you, but if i can support you please let me know 😊.

**Comment by XuesongYang** [comment]
Thanks for chiming in. I am refining the German TTS recipes, especially IPA g2p and tokenizations like preserving cases of the German alphabet and mixed inputs of graphemes/phonemes. They are coming up pretty soon.

w.r.t your latest dataset 22.10, the transcriptions in `metadata_*.csv` seem to echo twice. For example in `metadata_train.csv`, there are 3 columns separated by `|`, but the 2nd and 3rd columns seem to have exactly the same transcriptions. I could be wrong, but is it fine if only consider the 2nd column as the target transcriptions? 

```
5d4695c31da6b9305025048e5163d673|doch das soll sich nun ändern.|doch das soll sich nun ändern.
5dc6f7672bea5e68fcefd8a73abe514f|in der regierungsbefragung am nachmittag formulierte höferlin diesen punkt leicht anders.|in der regierungsbefragung am nachmittag formulierte höferlin diesen punkt leicht anders.
8a8699481570ee655c73cc28abf4493b|bei stehenden gewässern hat sich die lage vermutlich weniger gut entwickelt.|bei stehenden gewässern hat sich die lage vermutlich weniger gut entwickelt.
```

**Comment by thorstenMueller** [comment]
Hi @XuesongYang ,
thank you for working with my dataset.
Normally the 3rd column is like the 2nd, but normalized and in lower characters. As my input text has already been normalized (so numbers in written form) and in lower cases, the 2nd and 3rd column should be identically mostly.

**Comment by XuesongYang** [comment]
Good to know! Thanks. Maybe the 3rd column can be re-used to verify our German text normalization pipeline as well.

**Comment by XuesongYang** [comment]
This PR is the first attempt to refine the grapheme-only inputs tokenizer. I will push another PR with improvements using grapheme/phoneme mixed inputs tokenizer after this one is merged.

**Comment by thorstenMueller** [comment]
Before closing the issue i'd like to say thanks for adding 😊 and here's a link:
* https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/tts/datasets.html#thorsten-muller-s-german-neutral-tts-datasets

@XuesongYang Just being curious, but will a model be added here?:
https://catalog.ngc.nvidia.com/?filters=&orderBy=scoreDESC&query=tts

**Comment by XuesongYang** [comment]
@thorstenMueller Thank you for sharing datasets with the TTS society! Yes, the updated model for both datasets are here: https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/tts_de_fastpitchhifigan

**Comment by eqikkwkp25-cyber** [comment]
@XuesongYang Any update? Nemo 1.16.0 is out and models found here are not updated yet [https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/tts_de_fastpitchhifigan/files](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/tts_de_fastpitchhifigan/files)  though i dont know if the version number should always match.

**Comment by XuesongYang** [comment]
Yeah, we don't update models for every release of NeMo. But we will update the models once we have changed the recipe. We are refactoring TTS/G2P/Tokenizer a little bit so probably new models will be updated right after.
